### PR TITLE
[Atlantis] Add possibility to specify an ImagePullSecrets for pulling images from private registry

### PR DESCRIPTION
**Pull request by vlandemaine-orange**
Signed-off-by: Valentin Landemaine <valentin.landemaine.ext@orange.com>

# Helm Atlantis

#### EDIT: Fix also storageClass to storageClassName

@jkodroff 
@lkysow 
@jeff-knurek 
@sstarcher 

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/atlantis/templates/statefulset.yaml [hunk 1]
```diff
@@ -225,7 +229,7 @@ spec:
     spec:
       accessModes: ["ReadWriteOnce"] # Volume should not be shared by multiple nodes.
       {{- if .Values.atlantis_data_storageClass }}
-      storageClass: {{ .Values.atlantis_data_storageClass }} # Storage class of the volume
+      storageClassName: {{ .Values.atlantis_data_storageClass }} # Storage class of the volume
```
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml]: @sstarcher what do you think about renaming this: `.Values.atlantis_data_storageClass` -> `.Values.storageClassName`

File: stable/atlantis/templates/statefulset.yaml [hunk 2]
```diff
@@ -206,6 +206,10 @@ spec:
             mountPath: /etc/tls/
           {{- end }}
           resources:
+      {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+          - name: {{ .Values.imagePullSecrets }}
```
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml]:
```suggestion
      - name: {{ .Values.imagePullSecrets }}
```

**Comment by vlandemaine-orange** [comment]
I'v made the changes, it's ok fo you ?

**Comment by lkysow** [comment]
Please squash your commits into a single commit.

**Comment by vlandemaine-orange** [comment]
It's ok

**Comment by lkysow** [comment]
The DCO check is failing
image

**Comment by sstarcher** [comment]
Should we bump this to 1.X.0 since a breaking change was made.

**Comment by lkysow** [comment]
> Should we bump this to 1.X.0 since a breaking change was made.

Yes, we should. But then did you want to standardize the rest of the `values.yaml` to be camel case? If so, one of us can pull these commits into our own branch and also make those changes so there's a single PR with the 1.0.0 bump.

**Comment by sstarcher** [comment]
I'm fine with that if @vlandemaine-orange does not mind.

**Comment by vlandemaine-orange** [comment]
I made changes for switching all snake case variables to camel case and bump the version to 1.2.0. It's Ok for you ? I cannot made more commits, i don't have the time.

**Comment by lkysow** [comment]
We missed `serviceAccountSecrets.credentials-staging` and also removing the `atlantis` prefix from `atlantisDataStorage`.

Let's not be in too much of a rush to merge these PRs without thorough review please. The consequences of waiting a bit longer are worth it over having to bump versions often (in my opinion).

Also I didn't realize we were already at 1.0.0. This should be a BC-breaking bump to 2.0.0.

I will make these changes when I have some time.

**Comment by lkysow** [comment]
We also need to add an UPGRADING section to our readme as per https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#versioning

**Comment by vlandemaine-orange** [comment]
The chart is not working when imagePullSecrets is specified, i don't know why because it's working on my first commit..

Error: YAML parse error on atlantis/templates/statefulset.yaml: error converting YAML to JSON: yaml: line 89: mapping values are not allowed in this context

@lkysow

**Pull request merged**